meson.build

```diff
@@ -0,0 +1,48 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: run_command(['python', 'generate_version.py', '--print'], check: true).stdout().strip(),
+    license: 'BSD-3',
+    meson_version: '>=1.0.1',
+    default_options: [
+        # TODO: investigate, does meson try to compile against debug Python
+        # when buildtype = debug, this seems to be causing problems on CI
+        # where provided Python is not compiled in debug mode
+        'buildtype=release',
+        # TODO: Reactivate werror, some warnings on Windows
+        #'werror=true',
```

Comment:
Never add `-Werror` as a default flag in any project, that's a no no - instead put it in a CI job. I suggest deleting the above two lines.